File: vipclaw-admin/src/main/java/com/vipamp/vipclaw/admin/service/CaptchaService.java
package com.vipamp.vipclaw.admin.service;

import com.vipamp.vipclaw.admin.dto.CaptchaResponse;

/**
 * 验证码服务接口
 */
public interface CaptchaService {

    /**
     * 生成验证码
     *
     * @return 验证码响应，包含 Base64 图片和 key
     */
    CaptchaResponse generateCaptcha();

    /**
     * 验证验证码
     *
     * @param captchaKey 验证码 key
     * @param code       用户输入的验证码
     * @return 验证结果 true/false
     */
    boolean validateCaptcha(String captchaKey, String code);
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/service/CaptchaService.kt
package com.vipamp.vipclaw.admin.service

import com.vipamp.vipclaw.admin.dto.CaptchaResponse

/**
 * Captcha service interface
 */
interface CaptchaService {
    /**
     * Generate captcha
     *
     * @return Captcha response, contains Base64 image and key
     */
    fun generateCaptcha(): CaptchaResponse

    /**
     * Validate captcha
     *
     * @param captchaKey Captcha key
     * @param code       User input captcha code
     * @return Validation result true/false
     */
    fun validateCaptcha(captchaKey: String, code: String): Boolean
}
